### PR TITLE
Debug optimization bailouts

### PR DESCRIPTION
**Pull request by helfi92**
Add helpful debug output when using ModuleConcatenationPlugin.

Relevant documentation: https://webpack.js.org/plugins/module-concatenation-plugin/#debugging-optimization-bailouts

### CODE REVIEW

**Comment by edmorley** [comment]
Many thanks for doing this! I'd run out of time in #389 to try this one :-)

File: packages/library/index.js [hunk 1]
```diff
@@ -139,11 +139,18 @@ module.exports = (neutrino, opts = {}) => {
...
     })
-    .when(process.env.NODE_ENV === 'production', () => {
+    .when(process.env.NODE_ENV === 'production', (config) => {
       neutrino.use(minify);
-      neutrino.config
-        .plugin('module-concat')
-          .use(optimize.ModuleConcatenationPlugin);
+      config.plugin('module-concat').use(optimize.ModuleConcatenationPlugin);
+
+      if (neutrino.options.debug) {
```
Comment (eliperelman) [inline packages/library/index.js]: I think we'd want to do this regardless of being in `production`, right? If they are using `debug`, we should also use these I think...

Comment (helfi92) [reply packages/library/index.js]: Sure, we can remove the conditional. I'll send a patch now.

Comment (eliperelman) [reply packages/library/index.js]: Sorry, I think you may have misread my comment. We want these options to be set always when using debug, not always using production. It needs to come out of this `when` and go into its own when using `debug`.

Comment (helfi92) [reply packages/library/index.js]: I definitely misread the comment. Sorry for the confusion.

Comment (eliperelman) [reply packages/library/index.js]: No worries at all. 😃

File: packages/library/index.js [hunk 2]
```diff
@@ -131,6 +131,12 @@ module.exports = (neutrino, opts = {}) => {
           .end()
         .end()
       .end()
+    .merge({
```
Comment (eliperelman) [inline packages/library/index.js]: I think these are still supposed to be wrapped in a debug:

```js
.when(neutrino.options.debug, (config) => {
  config.merge({
    stats: {
      maxModules: Infinity,
      optimizationBailout: true
    }
  });
})
```

Comment (helfi92) [reply packages/library/index.js]: My bad...

**Pull request merged**